### DISCUSSION TITLE
Alembic dry run command

### DISCUSSION THREAD

**Discussion by carolguo-dd** [sqlalchemy/alembic#936] [Usage Questions]
**Describe the use case**

Is there a command that could print the the the raw sql command for any given command?
I find `--sql` not exactly the dry run command.
For example `alembic upgrade head --sql` prints out all the migration up to head, but it doesn't consider the current state of the databases. So the statements executed by `alembic upgrade head` isn't always the same as the output of `alembic upgrade head --sql`

**Databases / Backends / Drivers targeted**

**Example Use**

**Additional context**

**Have a nice day!**

**Comment by zzzeek** [comment]
you need to get the database's current version using "alembic current", then specify that in --sql, "alembic upgrade --sql <current_rev>:head".  --sql mode is intended to be useful when you dont have a database at all, so the two steps are separate.
